.ci/jobs/defaults.yml

```diff
@@ -0,0 +1,79 @@
+---
+
+##### GLOBAL METADATA
+
+- meta:
+    cluster: elasticsearch-ci
+
+##### JOB DEFAULTS
+
+- job:
+    vault:
+      url: "https://secrets.elastic.co:8200"
+      role_name: "elasticsearch-ci"
+    node: "rally"
+    concurrent: true
+    logrotate:
+      daysToKeep: 30
+      numToKeep: 500
+      artifactDaysToKeep: 7
+    parameters:
+      - string:
+          name: branch_specifier
+          default: "refs/heads/master"
+          description: "the Git branch specifier to build (&lt;branchName&gt;, &lt;tagName&gt;, &lt;commitId&gt;, etc.)\n"
+    scm:
+      - git:
+          name: origin
+          # master node jenkins user ~/.ssh
+          credentials-id: f6c7695a-671e-4f4f-a331-acdce44ff9ba
+          reference-repo: "/var/lib/jenkins/.git-references/rally.git"
+          branches:
+            - "${branch_specifier}"
+          url: "https://github.com/elastic/rally.git"
+          basedir: ""
+          wipe-workspace: true
+    triggers: []
+    wrappers:
+      - ansicolor
+      - timeout:
+          type: absolute
+          timeout: 80
+          fail: true
+      - timestamps
+    properties:
+      - github:
+          url: https://github.com/elastic/rally/
+    publishers:
+      - email:
+          recipients: infra-root+build@elastic.co
+      - junit:
+          allow-empty-results: true
+          results: 'junit-py*.xml'
+          keep-long-stdio: true
+      - google-cloud-storage:
+          credentials-id: 'elasticsearch-ci-gcs-plugin'
+          uploads:
+            - classic:
+                file-pattern: '.rally/$BUILD_NUMBER.tar.bz2'
+                storage-location: 'gs://elasticsearch-ci-artifacts/jobs/$JOB_NAME'
+                share-publicly: false
+                upload-for-failed-jobs: true
+                show-inline: true
+    builders:
+      - inject:
+          properties-content: |
+            HOME=$JENKINS_HOME
+            JAVA7_HOME=$HOME/.java/java7
+            JAVA8_HOME=$HOME/.java/java8
+            JAVA9_HOME=$HOME/.java/java9
+            JAVA10_HOME=$HOME/.java/java10
+            JAVA11_HOME=$HOME/.java/java11
+            JAVA12_HOME=$HOME/.java/openjdk12
+            JAVA13_HOME=$HOME/.java/openjdk13
+            JAVA14_HOME=$HOME/.java/openjdk14
+            JAVA15_HOME=$HOME/.java/openjdk15
+            JAVA16_HOME=$HOME/.java/openjdk16
+            JAVA17_HOME=$HOME/.java/openjdk17
+            ES_JAVA_HOME=$JAVA17_HOME
```

Comment:
Yep, I'll test this out. 

`version=${java##*openjdk}` is very nice BTW. :clap:

Reply:
Done in https://github.com/elastic/rally/pull/1561/commits/0114f53324b493cd3c581c743e70a6a538997c0c.

Note that we still set `ES_JAVA_HOME` directly to a specific Java version. We can't just default to the latest one available on disk, since we install newer JDKs on CI workers before we officially support them in order to get CI coverage ahead of time.

Reply:
> Done in [0114f53](https://github.com/elastic/rally/commit/0114f53324b493cd3c581c743e70a6a538997c0c).
> 
> Note that we still set `ES_JAVA_HOME` directly to a specific Java version. We can't just default to the latest one available on disk, since we install newer JDKs on CI workers before we officially support them in order to get CI coverage ahead of time.

Ok, I see.